source/common/runtime/runtime_impl.cc

```diff
@@ -444,23 +444,30 @@ void ProtoLayer::walkProtoValue(const ProtobufWkt::Value& v, const std::string&
   }
 }
 
-LoaderImpl::LoaderImpl(const ProtobufWkt::Struct& base, RandomGenerator& generator,
-                       Stats::Store& store, ThreadLocal::SlotAllocator& tls)
-    : LoaderImpl(DoNotLoadSnapshot{}, base, generator, store, tls) {
-  loadNewSnapshot();
-}
-
-LoaderImpl::LoaderImpl(DoNotLoadSnapshot /* unused */, const ProtobufWkt::Struct& base,
-                       RandomGenerator& generator, Stats::Store& store,
-                       ThreadLocal::SlotAllocator& tls)
-    : generator_(generator), stats_(generateStats(store)), admin_layer_(stats_), base_(base),
-      tls_(tls.allocateSlot()) {}
+LoaderImpl::LoaderImpl(Event::Dispatcher& dispatcher, ThreadLocal::SlotAllocator& tls,
+                       const envoy::config::bootstrap::v2::LayeredRuntime& config,
+                       absl::string_view service_cluster, Stats::Store& store,
+                       RandomGenerator& generator, Api::Api& api)
+    : generator_(generator), stats_(generateStats(store)), admin_layer_(stats_),
+      tls_(tls.allocateSlot()), config_(config), service_cluster_(service_cluster), api_(api) {
+  for (const auto& layer : config_.layers()) {
+    if (layer.has_admin_layer()) {
+      if (config_has_admin_layer_) {
+        throw EnvoyException(
+            "Too many admin layers specified in LayeredRuntime, at most one may be specified");
+      } else {
+        config_has_admin_layer_ = true;
```

Comment:
Is there anything in the docs about requiring exactly one admin layer?

Reply:
Yep, see https://github.com/envoyproxy/envoy/pull/7032/files#diff-0d35efbdc9c97a75473cdec333cfd2daR154